src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1129,6 +1207,7 @@
         {:outcome :not-found
          :message msg }))))
 
+;; TODO - Once we have the infra-crl functioning then this would need to be updated
```

Comment:
What will this return? Will it always return the infra-crl?

Reply:
How will this work with "intermediate CAs"

Reply:
Yes it would always return infra-crl - on the assumption that other compile masters do NOT use this to get the crl and only pure agent nodes would use it. They should be using the full CRL. I now wonder we will have a problem in case of a split install with console services, puppetdb - maybe not since agents do not talk to console. In some cases they actually might like when obtaining rbac token using a cli running from any agent node. But it takes in actual creds and does not rely on client cert so that should be ok. I will think more though. 

Correct me but here's my understanding of our "intermediate CA" support: PuppetCA is not a root but is issued by some other CA with an arbitrary level of hierarchy above. Obviously all entities within puppet infrastructure would need access to all intermediate issuers and their revocation status. PuppetCA, however, would only be responsible for all puppet certificates. So the infra-crl should be able to correctly report on status of all compile masters and MoM. 
That brings up a question (am suspecting an issue/gap presently with our intermediate CA support). All our certificate revocations need to be able to figure+get the revocation status for all issuers above PuppetCA. That may need to be provisioned in some manner either we adding an endpoint that serves these other CRLs (which puppetserver maintains uptodate copies of) or they be able to fetch CRLs themselves based on info in the intermediate CA certs (crlDistributionPoint) or do OCSP checking (AuthorityInfoAccess). The later two are possibilities depending on whatever PKI setup we become part of.

Reply:
Right now we're adding the ability for the cacrl to contain multiple crls in it. The `cacrl` setting would contain the crl that the CA adds and then also the crls for all of the authorities above it. The agent would then download a single file with multiple crls in it and respect each of them for full chain checking.
See https://github.com/puppetlabs/puppetserver/pull/1693 for that work in serverland.

Reply:
ok cool though that assumes the PKI can integrate with must support CRLs. Thats fair expectation I guess. I have not looked thru the above PR (will do) but who would be ensuring the combined CRL is kept up to date with all crls for CA's above us?

Reply:
Our assumption is that the crls are only for providing a chain of trust to the our CA and that the rest of the puppet infra is signed by that, so you'd only need to update it in the case of migrating/re-keying your CA, which should be very infrequent.

We're also working on a CLI tool to help with importing a chain when bootstrapping intermediate CAs which might be able to mitigate some of that pain.